src/main/java/lotto/LottoPrize.java

```diff
@@ -0,0 +1,34 @@
+package lotto;
+
+import java.util.Arrays;
+
+public enum LottoPrize {
+
+    NONE(0,0),
+    MATCH_3(3, 5000),
+    MATCH_4(4, 50000),
+    MATCH_5(5, 1500000),
+    MATCH_6(6, 2000000000);
+
+    private final Integer correctNumber;
+    private final Integer prize;
+
+    LottoPrize(Integer correctNumber, Integer prize) {
+        this.correctNumber = correctNumber;
+        this.prize = prize;
+    }
+
+    public static LottoPrize from(int count) {
+        return Arrays.stream(values())
+                .filter(lottoPrize -> lottoPrize.correctNumber == count)
+                .findFirst().orElse(LottoPrize.NONE);
+    }
```

Comment:
매 번 열거타입 인스턴스를 모두 순회하며 카운트 체크를 해야하는데, 캐싱을 활용하면 O(1)의 시간복잡도를 가지는 검색도 가능할 것 같네요!

```suggestion
    public static LottoPrize from(int count) {
        if (cache.containsKey(count)) {
            return cache.get(count);
        }
        return LottoPrize.NONE:
    }
```

Reply:
이런 부분은 신선했습니다! static block에서 초기화 한후 , 캐시를 도입해봤습니다.